sdks/python/apache_beam/runners/portability/fn_api_runner_transforms.py

```diff
@@ -200,7 +200,9 @@ def executable_stage_transform(
       stage_components.CopyFrom(components)
 
       # Only keep the referenced PCollections.
-      for pcoll_id in stage_components.pcollections.keys():
+      # Make pcollectionKey snapshot to avoid "Map modified during iteration"
+      # in py3
+      for pcoll_id in list(stage_components.pcollections.keys()):
```

Comment:
Thanks, @angoenka .
Could you please add a unit test that exercises this codepath? Our Py3 test suite would have caught it if there was a test that triggers this behavior.

Reply:
Added the tests. Had to dis